src/timeline/TimelineModel.cpp

```diff
@@ -260,6 +261,17 @@ TimelineModel::TimelineModel(TimelineViewManager *manager, QString room_id, QObj
         connect(&events, &EventStore::updateFlowEventId, this, [this](std::string event_id) {
                 this->updateFlowEventId(event_id);
         });
+
+        const auto userid = utils::localUser().toStdString();
```

Comment:
Global username is fetched here in timeline model instead of UserProfile because I attempted to fetch it in UserProfile and then emit a signal to notify that global username has been fetched but Qt again showed the different thread error. Couldn't use queued connection since QML dosen't allow that yet. So instead store the global username in timeline model and pass it to the UserProfile when I create it.

Is this okay? Any ideas on what else I could have done?

Reply:
You can use a queue connection, if that is done entirely in C++. You need 2 signals then though, one to change the value and one to show that it got changed.

Reply:
Oh ok I'll try shifting that code to the UserProfile class

Reply:
Ok done :D